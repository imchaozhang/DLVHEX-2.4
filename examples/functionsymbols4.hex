ft(X) :- &functionCompose[f,a](X).
decomp(F,A) :- ft(Y), &functionDecompose1[Y](F,A).
decomp2(F,A) :- ft(Y), &functionDecompose[Y,0](F), &functionDecompose[Y,1](A).
arity(A) :- ft(Y), &getArity[Y](A).
