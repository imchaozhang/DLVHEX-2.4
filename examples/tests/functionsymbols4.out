{ft(f(a)),decomp2(f,a),decomp(f,a),arity(1)}
